test/tools/ossfuzz/abiV2Proto.proto

```diff
@@ -0,0 +1,104 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+syntax = "proto2";
+
+// Flattened specification of array dimension
+// If is_static is false, and this array dimension is contained
+// inside another dimension e.g., x[][2] ([2] being the outer dimension)
+// then `length` for this dimension is the length of the first dynamically
+// sized array. The other (n-1) lengths are unspecified
+message ArrayDimensionInfo {
+    required uint32 length = 1;
+    required bool is_static = 2;
+}
+
+message ArrayType {
+    oneof base_type_oneof {
+        IntegerType inty = 1;
+        FixedByteType byty = 2;
+        AddressType adty = 3;
+        StructType stty = 4;
+    }
+    repeated ArrayDimensionInfo info = 5;
+}
+
+// uint8...256, int8...256
+message IntegerType {
+    required bool is_signed = 1;
+    required uint32 width = 2;
+}
+
+// bytes1, bytes2,..., bytes32
+message FixedByteType {
+    required uint32 width = 1;
+}
+
+// address, address payable
+message AddressType {
+    required bool payable = 1;
```

Comment:
the payable part should not make any difference for encoding...

Reply:
> Can you provide some example code produced by this routine?

I'm not sure which routine you are referring to here since I updated code since this review. Could you please repeat?

Reply:
I meant the whole thing. Can you provide some example code generated by the fuzzer?

Reply:
2-D array with 14 dynamically sized dimensions, code has changed since so please ignore the names of functions: https://gist.github.com/bshastry/5a9b5bbc8e5806c3694c41b23d0c2a88

5-D array with a 3 dynamically sized dimensions: https://gist.github.com/bshastry/f57087e9fbf579183a1859c1873fdfe5